platform/fabric/events_test.go

```diff
@@ -18,74 +17,88 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-type mockSubscriber struct {
-	listener events.Listener
-	m        sync.RWMutex
-}
-
-func (m *mockSubscriber) Subscribe(chaincodeName string, listener events.Listener) {
-	m.m.Lock()
-	defer m.m.Unlock()
-	m.listener = listener
-}
-
-func (m *mockSubscriber) Unsubscribe(chaincodeName string, listener events.Listener) {
-	m.m.Lock()
-	defer m.m.Unlock()
-	m.listener = nil
-}
-
-func (m *mockSubscriber) Publish(chaincodeName string, event *committer.ChaincodeEvent) {
-	m.m.RLock()
-	defer m.m.RUnlock()
-	if m.listener != nil {
-		m.listener.OnReceive(event)
-	}
-}
+const (
+	timeout = 75 * time.Millisecond
+	waitFor = 4 * timeout
+	tick    = timeout / 10
+)
 
 func TestEventListener(t *testing.T) {
 	subscriber := &mockSubscriber{}
 	listener := newEventListener(subscriber, "testChaincode")
-	eventChannel := listener.ChaincodeEvents()
+	ch := listener.ChaincodeEvents()
+
+	done := make(chan bool)
 
 	var wg sync.WaitGroup
+	wg.Add(2)
 
 	// Publish events
-	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		for i := 0; i < 1000; i++ {
-			subscriber.Publish("testChaincode", &committer.ChaincodeEvent{Payload: []byte(fmt.Sprintf("event %d", i))})
-			time.Sleep(1 * time.Millisecond)
+		msg := []byte("some msg")
+		for {
+			select {
+			case <-done:
+				return
+			default:
+				subscriber.Publish("testChaincode", &committer.ChaincodeEvent{Payload: msg})
```

Comment:
Without a sleep or a gosched, this will consume 100% of the CPU (unless `Publish` has a lot of blocking calls). I'd add either one of those.

Reply:
good catch - I've added a sleep